### PR TITLE
dev

### PR DESCRIPTION
**Pull request by mohankumar80**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (Rupamdas832) [inline public/index.html]: Change the title

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import "./styles.css";
+import { Routes, Route } from "react-router-dom";
+import Navigation from "./components/Navigation";
+
+import Home from "./components/Home";
+import ProductListing from "./components/ProductListing"
+import Cart from "./components/Cart";
+import WishList from "./components/WishList";
```
Comment (Rupamdas832) [inline src/App.js]: Make a separate index.js in components to export all the files from the components folder

File: src/components/ProductListing.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import React, { useReducer } from 'react'
+import { ProductsDB } from "./ProductsDB";
+import ProductCard from "./ProductCard";
+
+import "../styles.css";
+
+export default function ProductListing() {
+
+    function filterReducer(state, action) {
+        switch (action.type) {
```
Comment (Rupamdas832) [inline src/components/ProductListing.js]: You can make a separate file for reducer to clean up the product listing page

**Review by Rupamdas832 — CHANGES_REQUESTED** [review]
Some UI suggestions:
-Product-Card is too long for the mobile version. Too much scrolling
-In cart product-card could be a horizontal card 
-Some color-change you can do for Out-of-stock items to differentiate from others

Code LGTM

File: src/components/ProductCard.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            <div className="card-body">
+                <h3 className="card-title">{name}</h3>
+                <p className="card-text">by {author}</p>
+                <p className="card-text">---{category}</p>
+                <p className="card-text">Rs.{price}</p>
+                {
+                    cartItems.findIndex(item => item.id === id) === -1
+                        ? <button
+                            disabled={inStock ? false : true}
+                            style={inStock ? null : { cursor: "not-allowed" }}
```
Comment (neharajesh) [inline src/components/ProductCard.js:22]: Cursor style could be added as a class in your CSS.

File: src/context/reducer.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                } : item)
+            }
+        case "REMOVE_FROM_CART":
+            return { ...state, cartItems: state.cartItems.filter(item => item.id !== action.payload.id) }
+
+        /** wishlist functions*/
+        case "MOVE_TO_CART":
+            if (state.cartItems.findIndex(item => item.id === action.payload.id) === -1) {
+                return { ...state, cartItems: [...state.cartItems, action.payload], wishList: state.wishList.filter(item => item.id !== action.payload.id) }
+            } return { ...state, wishList: state.wishList.filter(item => item.id !== action.payload.id) }
```
Comment (neharajesh) [inline src/context/reducer.js]: I'm able to move stuff from wishlist to cart even if product is out of stock

File: src/context/reducer.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            }
+        case "DECREMENT_ITEMS":
+            if (action.payload.quantity === 1) {
+                return { ...state, cartItems: state.cartItems.filter(cartItem => cartItem.id !== action.payload.id) }
+            } return {
+                ...state, cartItems: state.cartItems.map(item => item.id === action.payload.id ? {
+                    ...item,
+                    quantity: item.quantity - 1
+                } : item)
+            }
```
Comment (neharajesh) [inline src/context/reducer.js]: When increasing or decreasing the quantity of items in cart, the number sometimes gets increased/decreased by 2 even on 1 click.

**Review by neharajesh — COMMENTED** [review]
Hey!
I found a small bug where the cart quantity changes by 2 on 1 click. 
And I can add an out of stock product to cart from wishlist.
Also, in the nav bar, the visited links turn blue with an underline, changing this would make your website look more amazing.
Apart from this, code looks good! 
Hope this helps!

File: public/index.html (hunk 1 above)
Comment (mohankumar80) [reply public/index.html]: Haven't observed it. Thanks for pointing out.

File: src/App.js (hunk 2 above)
Comment (mohankumar80) [reply src/App.js]: done

File: src/components/ProductListing.js (hunk 3 above)
Comment (mohankumar80) [reply src/components/ProductListing.js]: done

File: src/components/ProductCard.js (hunk 4 above)
Comment (mohankumar80) [reply src/components/ProductCard.js:22]: done

File: src/context/reducer.js (hunk 5 above)
Comment (mohankumar80) [reply src/context/reducer.js]: done

**Pull request merged**